x-pack/plugins/security_solution/server/lib/detection_engine/routes/rules/preview_rules_route.ts

```diff
@@ -184,6 +188,24 @@ export const previewRulesRoute = async (
               tags: [],
               updatedBy: rule.updatedBy,
             })) as TState;
+
+            // Save and reset error and warning logs
+            const currentLogs = {
+              errors: warningsAndErrorsStore
+                .filter((item) => item.newStatus === RuleExecutionStatus.failed)
+                .map((item) => item.message ?? 'ERROR'),
+              warnings: warningsAndErrorsStore
+                .filter(
+                  (item) =>
+                    item.newStatus === RuleExecutionStatus['partial failure'] ||
+                    item.newStatus === RuleExecutionStatus.warning
+                )
+                .map((item) => item.message ?? 'WARNING'),
```

Comment:
@marshallmain is there a good default to use for these messages that already exists? Would display when whatever error/warning occurred didn't have an associated message

Reply:
is it possible to have errors & warnings with no message? that's interesting, I think if that's the case we could probably handle it on the frontend with translated strings

Reply:
We shouldn't be creating errors or warnings with no message, so ideally we'd enforce that with type checking at some point. For now I think something like `Unknown error` and `Unknown warning` would be reasonable ways to represent it.